packages/amqp/lib/AbstractAmqpBaseConsumer.ts

```diff
@@ -50,25 +78,18 @@ export abstract class AbstractAmqpBaseConsumer<
     message: MessagePayloadType,
     messageType: string,
   ): Promise<Either<'retryLater', 'success'>> {
-    const barrierResult = await this.preHandlerBarrier(message, messageType)
+    const prehandlerOutput = await this.processPrehandlers(message, messageType)
+    const barrierResult = await this.preHandlerBarrier(message, messageType, prehandlerOutput)
```

Comment:
It would be great, but I doubt a fully universal solution is possible here. Imagine a prehandler with permission checks - if permissions changed since the last time we checked, prehandler results may be different.

Caching downstream on a case-by-case basis (e. g. how we cache user data in user plugin now) looks like a more feasible approach.

Reply:
I see, a fully universal solution doesn't seem to be the right approach you are right, so yeah, we would need to go with a case-by-case one 💯 but that is something for a future improvement :D